docs/tutorials/adding-a-module-to-your-runtime.md

```diff
@@ -317,44 +335,190 @@ To enable this, we simply need to add `Contracts` type that we defined in the `c
+
+/*** Add This Line ***/
+use contracts_rpc_runtime_api::ContractExecResult;
+/* --snip-- */
+```
+
+The next step, is to implement the required functions to handle the returned data.
+
+```rust
+impl_runtime_apis! {
+   /* --snip-- */
+
+   /*** Add This Block ***/
+    impl contracts_rpc_runtime_api::ContractsApi<Block, AccountId, Balance> for Runtime {
+        fn call(
+            origin: AccountId,
+            dest: AccountId,
+            value: Balance,
+            gas_limit: u64,
+            input_data: Vec<u8>,
+        ) -> ContractExecResult {
+            let exec_result = Contracts::bare_call(
+                origin,
+                dest.into(),
+                value,
+                gas_limit,
+                input_data,
+            );
+            match exec_result {
+                Ok(v) => ContractExecResult::Success {
+                    status: v.status,
+                    data: v.data,
+                },
+                Err(_) => ContractExecResult::Error,
+            }
+        }
+
+        fn get_storage(
+            address: AccountId,
+            key: [u8; 32],
+        ) -> contracts_rpc_runtime_api::GetStorageResult {
+            Contracts::get_storage(address, key).map_err(|rpc_err| {
+                use contracts::GetStorageError;
+                use contracts_rpc_runtime_api::{GetStorageError as RpcGetStorageError};
+                /// Map the contract error into the RPC layer error.
+                match rpc_err {
+                    GetStorageError::ContractDoesntExist => RpcGetStorageError::ContractDoesntExist,
+                    GetStorageError::IsTombstone => RpcGetStorageError::IsTombstone,
+                }
+            })
+        }
+    }
+   /*** End Added Block ***/
+}
+```
+
 ## Genesis Configuration
 
-The last thing we need to do in order to get your node up and running is to establish a genesis configuration for the Contracts pallet. Not all pallets will have a genesis configuration, but if they do, you can use its documentation to learn about it. For example, [`pallet_contracts::GenesisConfig` documentation](https://substrate.dev/rustdocs/master/pallet_contracts/struct.GenesisConfig.html) describes all the fields you need to define for the Contracts pallet. This definition is controlled in `substrate-node-template/src/chain_spec.rs`. We need to modify this file to include the `ContractsConfig` type and add the contract price units at the top:
+Not all pallets will have a genesis configuration, but if they do, you can use its documentation to learn about it. For example, [`pallet_contracts::GenesisConfig` documentation](https://substrate.dev/rustdocs/master/pallet_contracts/struct.GenesisConfig.html) describes all the fields you need to define for the Contracts pallet. This definition is controlled in `substrate-node-template/src/chain_spec.rs`. We need to modify this file to include the `ContractsConfig` type and the contract price units at the top:
```

Comment:
This bit about the GenesisConfig falls right in the middle of installing the RPC. What if you switch the order to.

* Exposing the contracts RPC API
* Install the RPC in the node's service (This is like the second half of exposing the RPC)
* Genesis Configuration